Sources/Sentry/SentrySDK.m

```diff
@@ -73,9 +73,6 @@ + (void)initialize
 + (SentryHub *)currentHub
 {
     @synchronized(self) {
-        if (nil == currentHub) {
-            currentHub = [[SentryHub alloc] initWithClient:nil andScope:nil];
-        }
```

Comment:
h:  We cant remove this right now. It will break a lot of tests, we need to find another solution.

Reply:
Could you explain motivation behind this?

Reply:
If it'll try to send envelopes to that hub, they'll be just dropped. And transport should wait for a hub with a client to send something.
May be we could check scope, before enrich it.

```
if (scope)
        [SentryDependencyContainer.sharedInstance.crashWrapper enrichScope:scope];
```

But that's strange solution and it could happen again in different part of code. Sentry should try to send something only when SentrySDK start finishes. With changes from this PR it's guaranteed.

Reply:
> Could you explain motivation behind this?

This was probably put in place to make writing tests easier. 
As you can see, all unit tests are broken right now. 
But I agree that we need to check if the SDK is initialized before starting to use the hub.